kerbauth.go

```diff
@@ -0,0 +1,129 @@
+package mssql
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/jcmturner/gokrb5/v8/client"
+	"github.com/jcmturner/gokrb5/v8/config"
+	"github.com/jcmturner/gokrb5/v8/credentials"
+	"github.com/jcmturner/gokrb5/v8/keytab"
+	"github.com/jcmturner/gokrb5/v8/spnego"
+)
+
+type krb5Auth struct {
+	username          string
+	realm             string
+	serverSPN         string
+	password          string
+	port              uint64
+	krb5Config        *config.Config
+	krbKeytab         *keytab.Keytab
+	krbCache          *credentials.CCache
+	initkrbwithkeytab bool
+	krb5Client        *client.Client
+	state             krb5ClientState
+}
+
+func getKRB5Auth(user, password, serverSPN string, krb5Conf *config.Config, keytabContent *keytab.Keytab, cacheContent *credentials.CCache, initkrbwithkeytab bool) (auth, bool) {
+	var port uint64
+	var realm, serviceStr string
+	var err error
+
+	params1 := strings.Split(serverSPN, ":")
+	if len(params1) != 2 {
+		return nil, false
+	}
+
+	params2 := strings.Split(params1[1], "@")
+	switch len(params2) {
+	case 1:
+		port, err = strconv.ParseUint(params1[1], 10, 16)
+		if err != nil {
+			return nil, false
+		}
+	case 2:
+		port, err = strconv.ParseUint(params2[0], 10, 16)
+		if err != nil {
+			return nil, false
+		}
+	default:
+		return nil, false
+	}
+
+	params3 := strings.Split(serverSPN, "@")
+	switch len(params3) {
+	case 1:
+		serviceStr = params3[0]
+		params3 = strings.Split(params1[0], "/")
+		params3 = strings.Split(params3[1], ".")
+		realm = params3[1] + "." + params3[2]
+	case 2:
+		realm = params3[1]
+		serviceStr = params3[0]
+	default:
+		return nil, false
+	}
+
+	return &krb5Auth{
+		username:          user,
+		serverSPN:         serviceStr,
+		port:              port,
+		realm:             realm,
+		krb5Config:        krb5Conf,
+		krbKeytab:         keytabContent,
+		krbCache:          cacheContent,
+		password:          password,
+		initkrbwithkeytab: initkrbwithkeytab,
+	}, true
+}
+
+func (auth *krb5Auth) InitialBytes() ([]byte, error) {
+	// Set to lookup KDCs in DNS
+	auth.krb5Config.LibDefaults.DNSLookupKDC = false
```

Comment:
1. Why is this set to false?
2. Why is this set to false here, rather then in the config parsing?
3. If set here, is it possible it will cause a race condition if owned elsewhere?